packages/pcd-types/src/pcd.ts

```diff
@@ -1,74 +1,165 @@
- * PCD, which allows the package some degree of control over how the PCD
- * is displayed in the passport application.
- */
-export interface DisplayOptions {
   /**
-   * Shown to the user in the main page of the passport, where they can
-   * see all of their cards. If `header` is undefined, the passport will use
-   * `renderCardBody` with `returnHeader` set to true.
+   * Human- or end-user-application-interpretable data.
    */
-  header?: string;
+  claim: C;
 
   /**
-   * Shown to the user in the `GenericProveScreen`, allowing them to
-   * disambiguate between different pcds of the same type. In the future,
-   * we'll have a better way to disambiguate between them.
+   * A cryptographic proof of the {@link PCD#claim}.
    */
-  displayName?: string;
+  proof: P;
 }
 
+/**
+ * Each type of {@link PCD} has a corresponding {@link PCDPackage}. The
+ * {@link PCDPackage} of a {@link PCD} type defines the code necessary to
+ * derive meaning from and operate on the data within a {@link PCD}.
+ *
+ * @typeParam {@link C} the type of {@link PCD.claim} for the {@link PCD} encapsulated
+ *   by this {@link PCDPackage}
+ *
+ * @typeParam {@link P} the type of {@link PCD.proof} for the {@link PCD} encapsulated
+ *   by this {@link PCDPackage}
+ *
+ * @typeParam {@link A} - the type of the arguments passed into {@link PCDPackage#prove}
+ *   to instantiate a new {@link PCD}. It is important that {@link A} can be serialized
+ *   and deserialized using {@code JSON.stringify} {@code JSON.parse}, because these arguments
+ *   should be able to be passed over the wire trivially. This may cause the type of {@link A}
+ *   to be less convenient that desired. Eg. you may have to pass {@code BigInt}s in as strings,
+ *   etc. Another important note about {@link A} is that each of its fields must implement the
+ *   {@link Argument} interface. This is important because it enables PCDpass to introspect the
+ *   arguments, and to implement useful features like the {@code GenericProveScreen}, which is
+ *   a screen that automatically build a UI which lets a user input all the arguments required to
+ *   instantiate a new instance of a particular {@link PCD} based on the request it gets from a
+ *   third party.
+ *
+ * @typeparam {@link I} the type of the arguments passed into {@link PCDPackage#init}, if the
+ *   init function is present to instantiate a new {@link PCD}
+ */
 export interface PCDPackage<C = any, P = any, A = any, I = any> {
+  /**
+   * The unique name identifying the type of {@link PCD} this package encapsulates.
+   */
   name: string;
+
+  /**
+   * Intended for use by PCDpass. Given a {@link PCD}, returns some information about
+   * how this {@link PCD} should be displayed to the user within the passport app.
+   */
   getDisplayOptions?: (pcd: PCD<C, P>) => DisplayOptions;
+
+  /**
+   * Intended to be used by PCDpass. Given a {@link PCD}, renders the body of a card
+   * that appears in PCDpass representing this {@link PCD}.
```

Comment:
This is a hack. It was initially introduced as a way for PCDs to control the way their header is rendered in PCDpass. I'll document more precisely in the code. I think we should get rid of it. Instead, we should have a notion of `PCDMetadata`. #187.

Right now, each PCD contains all of the information about itself.

I think we should introduce another abstraction that has some extra information. Eg. users should ostensibly be able to rename PCDs, but this is not currently possible because the display name of a PCD in PCDpass is some function of the PCD as defined in its PCD Package.

The metadata could include a bunch of other information, like the timestamp that this PCD was added to PCDpass.

Another thing it *could* contain is some 'pointer' to a render function. Eg. in your desktop operating system, you are able to have 'file associations' - there should be a way to specify that a particular instance of a PCD ought to be rendered differently from the other instances of this PCD. This need has come up a few painful places so far, and I expect it will continue to come up.

One example is that we've got a bunch of different types of signature PCDs (eddsa, rsa), and then for each type of thing that can be signed by each of these signatures, we also have a separate type of PCD. So there's an eddsa ticket PCD, and also an rsa image pcd.

Reply:
I think the functional suggestion in this comment was that you should explain the arguments/fields in the docs.  I appreciate the deeper dive, though, and the architectural questions this raises.

I like the analogy of file names, file types/extensions, and default apps.  That might be a way of looking at the question of different types of signed-data PCDs which we discussed offline.  The PCD type might be "rsa-signed-data" but the "file extension" equivalent might say that it's an image, or a ticket, etc.